lib/auth/scram.js

```diff
@@ -356,9 +355,13 @@ ScramSHA.prototype.auth = function(server, connections, db, username, password,
     );
   };
 
-  var _execute = function(_connection) {
-    process.nextTick(function() {
-      executeScram(_connection);
+  var _execute = function (_connection) {
+    process.nextTick(function () {
+      crypto.randomBytes(24, function (err, buff) {
+        if (err) throw err;
```

Comment:
Since this is now asynchronous, do not `throw` the error. Pass it to the callback and return.